x-pack/plugins/ingest_manager/server/collectors/package_collectors.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClient } from 'kibana/server';
+import _ from 'lodash';
+import { getPackageSavedObjects } from '../services/epm/packages/get';
+import { agentConfigService } from '../services';
+import { NewPackageConfig } from '../types';
+
+export interface PackageUsage {
+  name: string;
+  version: string;
+  enabled: boolean;
+}
+
+export const getPackageUsage = async (soClient?: SavedObjectsClient): Promise<PackageUsage[]> => {
+  if (!soClient) {
+    return [];
+  }
+  const packagesSavedObjects = await getPackageSavedObjects(soClient);
+  const agentConfigs = await agentConfigService.list(soClient, {
+    perPage: 1000, // avoiding pagination
+    withPackageConfigs: true,
+  });
+
+  // Once we provide detailed telemetry on agent configs, this logic should probably be moved
+  // to the (then to be created) agent config collector, so we only query and loop over these
+  // objects once.
+
+  // I couldn't convince TS that no undefined will end up in this string[][], so let it be any
+  const packagesInConfigs: any[][] = [];
+  agentConfigs.items.forEach((agentConfig) => {
+    const packageConfigs: NewPackageConfig[] = agentConfig.package_configs as NewPackageConfig[];
+    packagesInConfigs.push(
+      packageConfigs.map((pc) => pc.package?.name).filter((pn) => pn !== undefined)
+    );
+  });
```

Comment:
TS geeky hint: if you define the `.filter` function as a type guard function, TS will understand what you are achieving with it :)

```suggestion
  const packagesInConfigs = agentConfigs.items.map((agentConfig) => {
    const packageConfigs: NewPackageConfig[] = agentConfig.package_configs as NewPackageConfig[];
    return packageConfigs
      .map((pc) => pc.package?.name)
      .filter((pn): pn is string => pn !== undefined);
  });
```

Very verbose, but still maintaining type safeness :)

Reply:
Yay! Thank you :)